_posts/2020-06-04-App Service Continuous Deployment for Windows Containers with GitHub Actions.md

```diff
@@ -0,0 +1,403 @@
+# Use docker login
+- name: Docker Hub login
+  uses: azure/docker-login@v1
+  with:
+    username: ${{ secrets.DOCKERHUB_USERNAME }}
+    password: ${{ secrets.DOCKERHUB_PASSWORD }}
+```
+
+### Build and Push Image to Registry
+
+Under the same action, run the following docker build & push commands to publish to your chosen registry.  The "${{ github.sha }}" in the tags place will add in the github commit id so you know where it came from. Remember that your "container-name" is also defined here.
+
+You'll need the following secrets:
+
+- REGISTRY_USERNAME or DOCKERHUB_REGISTRY_NAME
+- IMAGE_NAME <br/>
+
+#### Azure Container Registry
+
+```yaml
+# Build and push the image to Azure Container Registry
+- name: Build and Push container to ACR
+  run: |
+    docker build --file=taskapp/taskapp/Dockerfile -t ${{ secrets.REGISTRY_USERNAME }}.azurecr.io/${{ secrets.IMAGE_NAME }}:${{ github.sha }} .
+    docker push ${{ secrets.REGISTRY_USERNAME }}.azurecr.io/${{ secrets.IMAGE_NAME }}:${{ github.sha }}
+```
+
+#### Docker Hub
+
+```yaml
+# Build and push the image to Azure Container Registry
+- name: Build and Push container to ACR
+  run: |
+    docker build --file=taskapp/taskapp/Dockerfile -t ${{ secrets.DOCKERHUB_REGISTRY_NAME }}/${{ secrets.IMAGE_NAME }}:${{ github.sha }} .
+    docker push ${{ secrets.DOCKERHUB_REGISTRY_NAME }}/${{ secrets.IMAGE_NAME }}:${{ github.sha }} 
+```
+
+Now that you have your image pushed to your registry.  You can push the container to App Service.  If you haven't already created your App Service, now is the time to do so before you can proceed.  Just remember that Windows Containers deployment is only available on the Premium SKU App Service Plans.  Be mindful of the SKU capacity levels as you may need to scale up if your container is too large.
+
+## Add Azure Authentication
+
+In order to automate deployment to App Serivce and Azure SQL, you'll need to use the credentials of the output from adding a Service Principal to authenticate your App Service.  
+
+You'll need the following secret:
+
+- AZURE_CREDENTIALS
+
+```yaml
+- name: Azure Service Principal Authentication
+  uses: azure/login@v1
+  with:
+      creds: ${{ secrets.AZURE_CREDENTIALS }}
+```
+
+## Deploy to Azure App Service
+
+The final step to setting up the continuous deployment is to add the webapps container deploy action.  
+
+You'll need the following secrets:
+
+- APP_NAME
+- REGISTRY_USERNAME
+- IMAGE_NAME
+
+### Azure Container Registry
+
+```yaml
+- name: Deploy Container to Azure App Service
+  uses: azure/webapps-container-deploy@v1
+  with:
+    app-name: ${{ secrets.APP_NAME }}
+    images: ${{ secrets.REGISTRY_USERNAME }}.azurecr.io/${{ secrets.IMAGE_NAME }}:${{ github.sha }}
+```
+
+### Docker Hub
+
+```yaml
+- name: Deploy Container to Azure App Service
+  uses: azure/webapps-container-deploy@v1
+  with:
+    app-name: ${{ secrets.APP_NAME }}
+    images: ${{ secrets.DOCKERHUB_REGISTRY_NAME }}/${{ secrets.IMAGE_NAME }}:${{ github.sha }}
```

Comment:
Same thing here, if we use the env var for the registry name, we can cut down on a lot of bloat

Reply:
```suggestion

```yaml
- name: Deploy Container to Azure App Service
  uses: azure/webapps-container-deploy@v1
  with:
    app-name: ${{ secrets.APP_NAME }}
    images: ${{ secrets.CONTAINER_REGISTRY_NAME }}/${{ secrets.IMAGE_NAME }}:${{ github.sha }}
    ```
```